### PR TITLE
[Projects] Add sorting, filtering and refresh

### PR DESCRIPTION
**Pull request by iurii-vatazhyshyn**
https://trello.com/c/dncU01ex/691-projects-add-sorting-filtering-and-refresh

### CODE REVIEW

File: src/common/Sort/Sort.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import { ReactComponent as Arrow } from '../../images/back-arrow.svg'
+
+import './sort.scss'
+
+const Sort = ({
+  isDescendingOrder,
+  onClickHandler,
+  options,
+  selectedValue,
```
Comment (mavdryk) [inline src/common/Sort/Sort.js:15]: rename to selectedId

File: src/common/Sort/Sort.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  setIsDescendingOrder
+}) => {
+  const [isOpenBody, setOpenBody] = useState(false)
+  const [selectedOption, setSelectedOption] = useState({})
+  const labelRef = useRef(null)
+  const arrowDirectionClassName = classNames(
+    isDescendingOrder && 'sort__down',
+    !isDescendingOrder && 'sort__up'
```
Comment (mavdryk) [inline src/common/Sort/Sort.js:23]: I believe you can use the ternary operator instead

File: src/components/ProjectsPage/projectsData.js [hunk 3]
```diff
@@ -55,5 +55,17 @@ export const generateProjectsStates = () => [
     label: 'Archived Projects'
   }
 ]
+export const generateProjectsSortOptions = () => [
+  {
+    id: 'byName',
+    label: 'By name',
+    sortPath: 'name'
```
Comment (mavdryk) [inline src/components/ProjectsPage/projectsData.js:62]: rename `sortPath` to `path`

File: src/components/ProjectsPage/Projects.js [hunk 4]
```diff
@@ -218,17 +224,69 @@ const Projects = ({
...
+            project.status.state === selectedProjectsState
+    },
+    [filterByName, selectedProjectsState]
+  )
+
+  const handleSortProject = useCallback(
+    (prevProject, nextProject) => {
+      const sortPath = projectsSortOptions.find(
+        option => option.id === sortProjectId
+      ).sortPath
```
Comment (mavdryk) [inline src/components/ProjectsPage/Projects.js:245]: move this code out of the iteration

File: src/components/ProjectsPage/Projects.js [hunk 5]
```diff
@@ -218,17 +224,69 @@ const Projects = ({
...
+
+      return isDescendingOrder
+        ? prevProject.metadata[sortPath] < nextProject.metadata[sortPath]
+          ? -1
+          : 1
+        : prevProject.metadata[sortPath] < nextProject.metadata[sortPath]
+        ? 1
+        : -1
```
Comment (mavdryk) [inline src/components/ProjectsPage/Projects.js:253]: use lodash method `orderBy`

File: src/components/ProjectsPage/Projects.js [hunk 6]
```diff
@@ -218,17 +224,69 @@ const Projects = ({
...
+
   useEffect(() => {
     setFilteredProjects(
-      projectStore.projects.filter(project => {
-        return (
-          (selectedProjectsState === 'allProjects' &&
-            project.status.state !== 'archived') ||
-          project.status.state === selectedProjectsState
+      projectStore.projects
+        .filter(project => handleFilterProject(project))
```
Comment (mavdryk) [inline src/components/ProjectsPage/Projects.js:261]:
```suggestion
        .filter(handleFilterProject)
```

File: src/components/ProjectsPage/Projects.js [hunk 7]
```diff
@@ -218,17 +224,69 @@ const Projects = ({
...
+  const handleFilterProject = useCallback(
+    project => {
+      return filterByName.length > 0
+        ? (project.metadata.name.includes(filterByName) &&
+            selectedProjectsState === 'allProjects' &&
+            project.status.state !== 'archived') ||
+            project.status.state === selectedProjectsState
+        : (selectedProjectsState === 'allProjects' &&
+            project.status.state !== 'archived') ||
+            project.status.state === selectedProjectsState
```
Comment (mavdryk) [inline src/components/ProjectsPage/Projects.js:236]: do not duplicate code

**Review by mavdryk — CHANGES_REQUESTED** [review]

File: src/common/Sort/Sort.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        </button>
+      </div>
+      {isOpenBody && (
+        <div className="sort__body" onClick={handleCloseSortBody}>
+          {options.map(option => {
+            return (
+              <SelectOption
+                item={option}
+                key={option.id}
+                onClick={id => onSelectOption(id)}
```
Comment (mariana-furyk) [inline src/common/Sort/Sort.js:75]:
```suggestion
                onClick={onSelectOption}
```

File: src/components/ProjectsPage/Projects.js [hunk 9]
```diff
@@ -218,17 +226,62 @@ const Projects = ({
...
+  }, [handleFilterProject, handleSortProjects, projectStore.projects])
+  // console.log(projectStore.projects)
+  useEffect(() => {
+    if (filterByName.length > 0) {
+      setFilterMatches(filteredProjects.map(func => func.metadata.name))
+    }
+  }, [filterByName, filteredProjects])
+
+  const handleSearchOnChange = value => {
+    if (value.length === 0) {
```
Comment (mariana-furyk) [inline src/components/ProjectsPage/Projects.js:274]: I think you can simplify this function and use just SetFilterByName :)

File: src/components/ProjectsPage/Projects.js [hunk 10]
```diff
@@ -243,19 +296,29 @@ const Projects = ({
...
       filteredProjects={filteredProjects}
+      filterMatches={filterMatches}
       handleCreateProject={handleCreateProject}
+      handleSearchOnChange={handleSearchOnChange}
+      isDescendingOrder={isDescendingOrder}
       isEmptyValue={isEmptyValue}
       match={match}
       nuclioStore={nuclioStore}
       projectStore={projectStore}
+      projectsSortOptions={projectsSortOptions}
```
Comment (mariana-furyk) [inline src/components/ProjectsPage/Projects.js:309]: you can declare options in View component and if it's static data you don't need useMemo

**Pull request closed without merging**